### PR TITLE
Update scene.py

### PR DESCRIPTION
**Pull request by PlayfulMathematician**
## Overview: What does this pull request change?

This adds a clear function to manim.Scene that clears the screen

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/scene/scene.py [hunk 1]
```diff
@@ -146,6 +146,9 @@ def __init__(
             random.seed(self.random_seed)
             np.random.seed(self.random_seed)
 
+    def clear():
```
Comment (behackl) [inline manim/scene/scene.py:149]: Methods need `self` as an argument, and we can also document the return type:

```suggestion
    def clear(self) -> None:
```

File: manim/scene/scene.py [hunk 2]
```diff
@@ -146,6 +146,9 @@ def __init__(
...
+        self.remove(*self.mobjects)
```
Comment (behackl) [inline manim/scene/scene.py:150]: Is it actually necessary to call `Scene.remove`, or would it be sufficient to simply set `self.mobjects = []`? The latter would be more performant, perhaps it is worth a try.

```suggestion
        self.remove(*self.mobjects)
```

**Review by behackl — CHANGES_REQUESTED** [review]
Hey, thanks for your contribution! I left some suggestions that you could apply directly, but in order to get this merged, there are two more things that you have to provide:

- [ ] A docstring, in which the functionality of the new method is documented (see, for example, the docstring of `Scene.construct` for guidance), and
- [ ] at least one test that makes sure that the functionality works as expected (take a look at `tests/module/scene/test_scene.py` to see what I mean).

**Comment by PlayfulMathematician** [comment]
I have made the changes

**Review by behackl — CHANGES_REQUESTED** [review]
The way the additional functionality is implemented is *not a good idea*: effectively, you are modifying a list while you are iterating over it, which leads to *unexpected* results:
```py
>>> nums = [k for k in range(20)]
>>> for k in nums:
...     if k < 10:
...         nums.remove(k)
...
>>> nums
[1, 3, 5, 7, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
```
You would need to implement this in a different way (e.g., by first identifying all mobjects to be removed, and then removing them in a separate call / outside of the iteration over `self.mobjects`).

Besides that, I'm not sure I find it very intuitive to have this sort of functionality in a `Scene.clear` method; semantically I'd expect that `clear` simply does one thing, namely *clear* the scene. But regardless, I won't block the PR over this feature; if you feel it makes sense, it can be included.

For the sake of readability, I recommend to implement with a default value of `None`, and special case the removal of mobjects in that case simply as `self.mobjects = []`; i.e.,

```py
def clear(self, condition: Callable[[Mobject], bool] | None = None) -> None:
    if condition is None:
        ...
```

Also, we definitely require tests for a more complex functionality like this.

**Pull request closed without merging**